embed/config.go

```diff
@@ -381,8 +381,8 @@ func (cfg *Config) UpdateDefaultClusterFromName(defaultInitialCluster string) {
 		// if client-listen-url is 0.0.0.0, just use detected default host
 		// otherwise, rewrite advertise-client-url with localhost
 		if ip != "0.0.0.0" {
-			_, acPort, _ := net.SplitHostPort(cfg.ACUrls[0].Host)
-			cfg.ACUrls[0] = url.URL{Scheme: cfg.ACUrls[0].Scheme, Host: fmt.Sprintf("localhost:%s", acPort)}
+			ahost, acPort, _ := net.SplitHostPort(cfg.ACUrls[0].Host)
```

Comment:
This PR (https://github.com/coreos/etcd/pull/7027/files) does the wrong thing. We should not blindly rewrite the advertise url to localhost just because the ip is not 0.0.0.0. In another word, even if cfg.IsDefaultHost() returns true, it does not mean that we have overwritten both client and peer advertise urls. That is why we might overwrite the users configuration.

Please make sure you fully understand what happened before making the change.